src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -187,6 +187,15 @@ public override int Timeout
             set { this.httpRequest.Timeout = (int)Math.Min(Int32.MaxValue, new TimeSpan(0, 0, value).TotalMilliseconds); }
         }
 
+        /// <summary>
+        /// Gets or sets the read and write timeout (in seconds) for this request.

```

Comment:
Although this says it is in seconds, the set value isn't the same value that is read out. 
if i set to 100s when reading it seems like the returned value will be in milliseconds

Reply:
@marabooy yes you set the value in seconds.. but when it is being passed to HttpWebRequest it is converted to milliseconds.

Reply:
This is not consistent and would be a source of bugs :-(
```
var message = new HttpWebRequestMessage();
var expected = 1; // second
message.ReadWriteTimeout = expected;
var actual = message.ReadWriteTimeout; // => 1000ms?!
Assert.AreEqual(expected, actual, "Property set should generally be equal to property read");
```
Options for this could be
- Use various properties for millisecond get/set, second get/set, minutes get/set... Or simply change the type to a Timespan
- Use ```SetReadWriteTimeoutInSeconds(int seconds)``` and ```GetReadWriteTimeoutInMilliseconds()```
- ?

Reply:
Ok. so the ReadWriteTimeout property has been added to the <code> DataServiceContext </code> class. This means that when a user is configuring the context that is where the set a value for this property in seconds(it is easier for users). But within the code when this value is being set to <code> HttpWebRequest</code> it is converted to milliseconds. This means that what the <HttpWebRequest </code> gets is in milliseconds. In your case there, that is not how it works. 
<pre>
var valfromcontext = 1; //second
var message = new HttpWebRequestMessage();
var expected = 1000; //millisecond when passing to webrequestmessage we convert this value to milliseconds
message.ReadWriteTimeout = expected;
var actual = message.ReadWriteTimeout; // => 1000ms?!
Assert.AreEqual(expected, actual, "Property set should generally be equal to property read");

</pre>